### PR TITLE
add libsigcpp/3.0.0

### PR DESCRIPTION
**Pull request by gunmetal313**
Specify library name and version: libsigcpp/3.0.0

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Not finished. The problem with include

### CODE REVIEW

**Comment by gunmetal313** [comment]
c:\projects\conan-center-index\recipes\libsigcpp\3.0.0\test_package\test_package.cpp(11): fatal error C1083: Cannot open include file: 'sigc++/sigc++.h':

File: recipes/libsigcpp/3.0.0/0001-libsigcpp.patch [hunk 1]
```diff
@@ -0,0 +1,41 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (Croydon) [inline recipes/libsigcpp/3.0.0/0001-libsigcpp.patch]: Did you propose this patch upstream?

File: recipes/libsigcpp/3.0.0/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
```
Comment (uilianries) [inline recipes/libsigcpp/3.0.0/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/3.0.0/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+# -*- coding: utf-8 -*-
```
Comment (uilianries) [inline recipes/libsigcpp/3.0.0/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/3.0.0/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+import os
+from conans import ConanFile, CMake, tools
+
+class LibSigCppConan(ConanFile):
+    name = "libsigcpp"
+    version = "3.0.0"
+    url = "https://github.com/libsigcplusplus/libsigcplusplus"
```
Comment (uilianries) [inline recipes/libsigcpp/3.0.0/conanfile.py]:
```suggestion
    url = "https://github.com/conan-io/conan-center-index"
```

File: recipes/libsigcpp/3.0.0/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    homepage = "https://libsigcplusplus.github.io/libsigcplusplus/"
+    author = "Conan Community"
+    license = "LGPL"
```
Comment (uilianries) [inline recipes/libsigcpp/3.0.0/conanfile.py]:
```suggestion
    license = "LGPL-3.0"
```

File: recipes/libsigcpp/3.0.0/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+import os
+from conans import ConanFile, CMake, tools
+
+class LibSigCppConan(ConanFile):
+    name = "libsigcpp"
+    version = "3.0.0"
+    url = "https://github.com/libsigcplusplus/libsigcplusplus"
+    homepage = "https://libsigcplusplus.github.io/libsigcplusplus/"
+    author = "Conan Community"
```
Comment (uilianries) [inline recipes/libsigcpp/3.0.0/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/3.0.0/test_package/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
```
Comment (uilianries) [inline recipes/libsigcpp/3.0.0/test_package/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/3.0.0/test_package/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+# -*- coding: utf-8 -*-
```
Comment (uilianries) [inline recipes/libsigcpp/3.0.0/test_package/conanfile.py]:
```suggestion
```

File: recipes/libsigcpp/config.yml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+versions:
```
Comment (uilianries) [inline recipes/libsigcpp/config.yml]: this file is not required, as you are not using the generic folder `all`

File: recipes/libsigcpp/3.0.0/test_package/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,8 @@
+PROJECT(test_package CXX)
```
Comment (uilianries) [inline recipes/libsigcpp/3.0.0/test_package/CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 3.2)
```

File: recipes/libsigcpp/3.0.0/test_package/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,8 @@
...
+cmake_minimum_required(VERSION 3.2)
```
Comment (uilianries) [inline recipes/libsigcpp/3.0.0/test_package/CMakeLists.txt]:
```suggestion
PROJECT(test_package CXX)
```

**Comment by uilianries** [comment]
@gunmetal313 please, take a look https://github.com/gunmetal313/conan-center-index/pull/1

**Comment by uilianries** [comment]
hmm, it requires C++17

**Comment by uilianries** [comment]
@gunmetal313 https://github.com/gunmetal313/conan-center-index/pull/2

**Comment by Croydon** [comment]
I know nothing about this library but the Bincrafters version seem to have static support https://github.com/bincrafters/conan-sigcpp

**Comment by danimtb** [comment]
There are still some small things to improve in this recipe but as the PR has been open for a long time we can merge it and improve it later. I will open an issue first

**Comment by danimtb** [comment]
Issue open to address the missing porints on this PR: #521 Thanks!

File: recipes/libsigcpp/3.0.0/test_package/conanfile.py (hunk 7 above)
Comment (danimtb) [inline recipes/libsigcpp/3.0.0/test_package/conanfile.py]: remove line

File: recipes/libsigcpp/3.0.0/test_package/conanfile.py (hunk 8 above)
Comment (danimtb) [inline recipes/libsigcpp/3.0.0/test_package/conanfile.py]: remove line

File: recipes/libsigcpp/config.yml (hunk 9 above)
Comment (danimtb) [reply recipes/libsigcpp/config.yml]: Use a generic folder

File: recipes/libsigcpp/3.0.0/conanfile.py [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Linux":
+            self.cpp_info.libs.append("m")
```
Comment (danimtb) [inline recipes/libsigcpp/3.0.0/conanfile.py]:
```suggestion
            self.cpp_info.system_libs.append("m")
```

**Pull request merged**